File: Backend/src/main/java/com/unesco/core/entities/schedule/Pair.java
package com.unesco.core.entities.schedule;

import javax.persistence.*;

@Entity
@Table(name = "un_pair")
public class Pair {
    @Id
    @SequenceGenerator(name = "pairSequenceGen", sequenceName = "pairSequenceGen", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "pairSequenceGen")
    private long id;
    private int pairNumber;
    private String dayofweek;
    private String weektype;
    @ManyToOne
    @JoinColumn(name = "room_id", referencedColumnName = "id")
    private Room room;
    @ManyToOne
    @JoinColumn(name = "lesson_id", referencedColumnName = "id")
    private Lesson lesson;

    public Pair() {
    }

    public long getId() {
        return id;
    }
    public void setId(long id) {
        this.id = id;
    }

    public int getPairNumber() {
        return pairNumber;
    }
    public void setPairNumber(int pairNumber) {
        this.pairNumber = pairNumber;
    }

    public String getDayofweek() {
        return dayofweek;
    }
    public void setDayofweek(String dayofweek) {
        this.dayofweek = dayofweek;
    }

    public String getWeektype() {
        return weektype;
    }
    public void setWeektype(String weektype) {
        this.weektype = weektype;
    }

    public Room getRoom() {
        return room;
    }
    public void setRoom(Room room) {
        this.room = room;
    }

    public Lesson getLesson() {
        return lesson;
    }
    public void setLesson(Lesson lesson) {
        this.lesson = lesson;
    }

}

File: Frontend/src/app/models/shedule/pair.ts
import {Room} from "./room.model";
import {Lesson} from "./lesson";
import {PairType} from "./pairType";

export class Pair {
    public id: number;
    public pairNumber: number;
    public dayofweek: String;
    public weektype: String;
    public room: Room;
    public pairType: PairType;
    public lesson: Lesson;

    constructor() {
        this.id = 0;
        this.pairNumber = 0;
        this.weektype = "";
        this.dayofweek = "";
        this.lesson = new Lesson();
        this.pairType = new PairType();
        this.room = new Room();
    }
}
